src/components/inputs/rich-text-input/rich-text-input.js

```diff
@@ -10,16 +9,40 @@ import html from '../../internals/rich-text-utils/html';
 import isEmpty from '../../internals/rich-text-utils/is-empty';
 
 class RichTextInput extends React.PureComponent {
+  state = {
+    // we keep track of the serialized (HTML) value
+    serializedValue: this.props.value || '',
+    value: html.deserialize(this.props.value || ''),
+  };
+
+  componentDidUpdate() {
+    // if the value provided is not in sync with the RichTextInput,
+    // then reset our value by deserializing this new value
+    if (this.props.value !== this.state.serializedValue) {
```

Comment:
How much can this just be a memoized value instead of keeping the syncing in a lifecycle.